test/FunctionalTests/Microsoft.OData.Core.Tests/Build.NetFramework/TestMetadataSelector.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Vocabularies;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Evaluation
+{
+    public class TestMetadataSelector : ODataMetadataSelector
+    {
+        public TestMetadataSelector()
+        {
+        }
+
+        public string PropertyToOmit;
+
+        public IEnumerable<IEdmNavigationProperty> NavigationPropertyToAdd;
+
+        /// <summary>
+        /// Determines which navigation links should be selected.
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="navigationProperties"></param>
+        /// <returns></returns>
+        public override IEnumerable<IEdmNavigationProperty> SelectNavigationProperties(IEdmStructuredType type, IEnumerable<IEdmNavigationProperty> navigationProperties)
+        {
+            if (NavigationPropertyToAdd != null)
+            {
+                return NavigationPropertyToAdd;
+            }
+            return navigationProperties.Where(s => s.Name != PropertyToOmit);
+        }
+
+        public override  IEnumerable<IEdmOperation> SelectBindableOperations(IEdmStructuredType type, IEnumerable<IEdmOperation> bindableOperations)
+        {
+            //Omit all actions

```

Comment:
Actually, this omits all operations not named "HasHat", right?

Reply:
This was a left over comment. This selects only the operations named HasHat.

---
In reply to: [310766305](https://github.com/OData/odata.net/pull/1494#discussion_r310766305) [](ancestors = 310766305)